kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/LoadAsTemplateTest.java

```diff
@@ -0,0 +1,89 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.client.mock;
+
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import io.fabric8.kubernetes.api.builder.TypedVisitor;
+import io.fabric8.kubernetes.api.model.EnvVarBuilder;
+import io.fabric8.kubernetes.api.model.KubernetesList;
+import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
+import io.fabric8.openshift.client.DefaultOpenShiftClient;
+import io.fabric8.openshift.client.OpenShiftClient;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class LoadAsTemplateTest {
+
+  @Test
+  public void shouldLoadPodAsTemplate() throws Exception {
+    OpenShiftClient client = new DefaultOpenShiftClient();
+    KubernetesList list = client.templates().load(getClass().getResourceAsStream("/test-pod-create-from-load.yml")).processLocally();
+    assertNotNull(list);
+    assertNotNull(list.getItems());
+    assertEquals(1, list.getItems().size());
+  }
+
+  @Test
+  public void shouldProcessLocally() throws Exception {
+    OpenShiftClient client = new DefaultOpenShiftClient();
+    Map<String, String> map = new HashMap<>();
+    map.put("USERNAME", "root");
+
+    KubernetesList list = client.templates().load(getClass().getResourceAsStream("/template-with-params.yml")).processLocally(map);
+    assertListIsProcessed(list);
+  }
+
+  @Test
+  public void shouldProcessLocallyWithParametersInYaml() throws Exception {
+    OpenShiftClient client = new DefaultOpenShiftClient();
+    Map<String, String> map = new HashMap<>();
+    map.put("USERNAME", "root");
+
+    KubernetesList list = client.templates().load(getClass().getResourceAsStream("/template-with-params.yml")).processLocally(getClass().getResourceAsStream("/parameters.yml"));
+    assertListIsProcessed(list);
+  }
+
+  //Check that the processed template is as expected
+  private void assertListIsProcessed(KubernetesList list) {
+    assertNotNull(list);
+    assertNotNull(list.getItems());
+    assertEquals(1, list.getItems().size());
+
+    final AtomicBoolean userIsRoot = new AtomicBoolean(false);
+    final AtomicBoolean passwordIsNotNull = new AtomicBoolean(false);
+    new KubernetesListBuilder(list).accept(new TypedVisitor<EnvVarBuilder>() {
+
+      @Override
+      public void visit(EnvVarBuilder element) {
+        if (element.getName().equals("USERNAME")) {
+          userIsRoot.set(element.getValue().equals("root"));
+        } else if (element.getName().equals("PASSWORD")) {
+          passwordIsNotNull.set(element.getValue() != null);
```

Comment:
Check length too perhaps to make sure it's got a generated value, not just empty string?